packages/cli-platform-android/src/commands/buildAndroid/startServerInNewWindow.ts

```diff
@@ -6,8 +6,12 @@ import {logger} from '@react-native-community/cli-tools';
 export function startServerInNewWindow(
   port: number,
   terminal: string,
-  reactNativePath: string,
+  projectRoot: string,
 ) {
+  const reactNativeCliPath = path.dirname(
```

Comment:
This is going to introduce all sorts of issues in monorepos. In the following example, can you guess which version of `@react-native-community/cli-plugin-metro` gets used?
```
root
├── node_modules
│   └── @react-native-community/cli-metro-plugin <-- which version got hoisted here?
├── package.json
└── packages
    ├── app-one
    │   ├── node_modules
    │   │   └── react-native@0.70
    │   ├── package.json
    │   └── src
    └── app-two
        ├── node_modules
        │   └── react-native@0.71
        ├── package.json
        └── src
```

Reply:
It's also relying on hoisting. Not all package managers do this. If you're using pnpm, they transitive dependencies are only available for the package that depended on them. With this solution, the project is required to add `@react-native-community/cli-plugin-metro` in their own `package.json` for this to work. That's a breaking change, and it adds unnecessary burden on the app author.

Reply:
But do we have any other options? Right now in monorepos when launching `npx react-native start` from e.g. `app-one` it will start packager from root of monorepo, so `app-one` won't bundle correctly. With this change only monorepos that contains apps with different RN version will have to add this pacakge `@react-native-community/cli-plugin-metro` to `package.json`.

Reply:
> Right now in monorepos when launching `npx react-native start` from e.g. `app-one` it will start packager from root of monorepo, so `app-one` won't bundle correctly.

This sounds like a bug. Do we have a repro for this? `npx react-native start` should start with the `local node_modules/.bin` before retrying from parent directories. If you ran this command under `packages/app-one` and it has different version of `react-native` from the rest of the repository, `npx` should find `packages/app-one/node_modules/.bin/react-native`.

Reply:
> Do we have a repro for this? 

Yeah, we have repro [here](https://github.com/MateWW/react-native-new-arch-monorepo/). When you launch `npx react-native run-ios` or `npx react-native run-android` inside `WorldHello` or `HelloWorld` metro bundler will launch from root of monorepo. I previously wrote about `npx react-native start`, but it's about `run-ios` and `run-android` my mistake, sorry 😅

Reply:
I've had a quick look at how `run-android` launches the packager, and I agree with @cortinico. You should not call the [`launchPackager`](https://github.com/facebook/react-native/blob/main/scripts/launchPackager.command) (or the [underlying script](https://github.com/facebook/react-native/blob/main/scripts/packager.sh) it runs). They're both assuming that `react-native` is always installed within the project root. In a monorepo, this isn't always true.

The script ends up calling `"$REACT_NATIVE_ROOT/cli.js" start --custom-log-reporter-path "$THIS_DIR/packager-reporter.js" "$@"`. Can we call this directly instead? It seems to be doing the right thing when I call it manually.

Reply:
Yes, but there are limitation with `open` command, if we're running `run-ios` or `run-android` a new terminal window is opened and that's why I created `launchPackeger.command` & `.packager.env` inside CLI, because we can't unfortunately pass command with arguments to new created window.  See here: https://github.com/react-native-community/cli/pull/1812#discussion_r1092434402 But with these files I can store information in `.packager.env` and launch packager in new terminal window with `launchPackager.command`.

Reply:
Can we pass `--args` and/or `--env` options to `open`?
![image](https://user-images.githubusercontent.com/4123478/222425983-afc84aba-9990-492b-960f-46b37a23e985.png)

Reply:
Sorry, I had assumed that when it says `launchPackager.command` here, that you're referring to `react-native/scripts/launchPackager.command`. I missed that you had created your own. 🤦‍♂️

This is looks fine. The only objection is trying to resolve `@react-native-community/cli-plugin-metro` from this package. As mentioned earlier, this assumes hoisting and you might pick up the wrong version. For this to work correctly, I think we have to do one of two things:

1. Add `@react-native-community/cli-plugin-metro` as a dependency of `@react-native-community/cli-platform-android`. This will ensure that `cli-platform-android` gets the correct copy linked into its `node_modules` folder if necessary.
2. Resolve `@react-native-community/cli-plugin-metro` via `@react-native-community/cli`. This is a bit more frail since it assumes that `cli-platform-android` was installed via `cli`, which may be okay here.

Reply:
I think the first solution you proposed is better. 👍

Reply:
But there's a problem when we add `@react-native-community/cli-plugin-metro` to `package.json`, and we try to find this package via `require.resolve('@react-native-community/cli-plugin-metro/package.json')` it finds this package inside `node_modules` but in CLI (`/Users/szymonrybczak/callstack/forked-cli/packages/cli-platform-android/node_modules/@react-native-community/cli-plugin-metro/build/launchPackager.command`).

Reply:
Oh, does the version number match what is used in `@react-native-community/cli`? If the two match, they should be deduped. I don't think you should see a separate copy there.